subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -162,6 +164,39 @@ public boolean shouldWatch(FileSystemLocationSnapshot snapshot) {
         return !ignoredForWatching(snapshot) && isInWatchableHierarchy(snapshot.getAbsolutePath());
     }
 
+    public void updateUnwatchableFileSystems(WatchMode watchMode) {
+        unwatchableHierarchies = watchUnsupportedFileSystems(watchMode)
+            ? DefaultFileHierarchySet.of()
+            : detectUnsupportedFileSystems();
+    }
+
+    private boolean watchUnsupportedFileSystems(WatchMode watchMode) {
+        return watchMode != WatchMode.DEFAULT;
+    }
+
+    private FileHierarchySet detectUnsupportedFileSystems() {
+        return forEachUnsupportedFileSystem(
+            DefaultFileHierarchySet.of(),
+            (fileHierarchySet, fileSystemInfo) -> fileHierarchySet.plus(fileSystemInfo.getMountPoint()),
+            "Assuming all file systems can be watched",
+            DefaultFileHierarchySet.of()
+        );
+    }
+
+    private <T> T forEachUnsupportedFileSystem(T current, BiFunction<T, ? super FileSystemInfo, T> accumulator, String failureAction, T onFailure) {
+        try {
+            return watchableFileSystemDetector.detectUnsupportedFileSystems()
+                .reduce(
+                    current,
+                    accumulator,
+                    nonCombining()
+                );
```

Comment:
I think this could fit on one line now

Reply:
I inlined the method, so this is not relevant any more.